### PR TITLE
StatusStore migration

### PR DESCRIPTION
**Pull request by rvillablanca**
#### Summary

Migration of `StatusStore` to return plain errors.

Fixes https://github.com/mattermost/mattermost-server/issues/14814

### CODE REVIEW

File: store/sqlstore/status_store.go [hunk 1]
```diff
@@ -66,72 +63,61 @@ func (s SqlStatusStore) Get(userId string) (*model.Status, *model.AppError) {
...
 	}
 
 	return statuses, nil
 }
 
-func (s SqlStatusStore) ResetAll() *model.AppError {
+func (s SqlStatusStore) ResetAll() error {
 	if _, err := s.GetMaster().Exec("UPDATE Status SET Status = :Status WHERE Manual = false", map[string]interface{}{"Status": model.STATUS_OFFLINE}); err != nil {
-		return model.NewAppError("SqlStatusStore.ResetAll", "store.sql_status.reset_all.app_error", nil, "", http.StatusInternalServerError)
+		return errors.Wrap(err, "failed to update States")
```
Comment (agnivade) [inline store/sqlstore/status_store.go:104]: typo at Status

File: store/sqlstore/status_store.go [hunk 2]
```diff
@@ -66,72 +63,61 @@ func (s SqlStatusStore) Get(userId string) (*model.Status, *model.AppError) {
...
 		Where(sq.Eq{"UserId": userIds})
 	queryString, args, err := query.ToSql()
 	if err != nil {
-		return nil, failure(err)
+		return nil, errors.Wrap(err, "status_tosql")
 	}
 	rows, err := s.GetReplica().Db.Query(queryString, args...)
 	if err != nil {
-		return nil, failure(err)
+		return nil, errors.Wrap(err, "failed to find States")
```
[the remark this answers is not in the document]
Comment (rvillablanca) [reply store/sqlstore/status_store.go:84]: That was intentional, I try to use plural wen you use `Select` and singular when you use `SelectOne`, although now not sure if `states` is ok or it should be `statuses` :man_shrugging:. Do you think we should use always the singular name entity?

[the remark this answers is not in the document]
Comment (agnivade) [reply store/sqlstore/status_store.go:84]: Oh I see. Yeah, I think `Statuses` is more clear. Thanks.

**Comment by agnivade** [comment]
Some new code got in master which uses the status store in app/server.go. That needs to be fixed.

**Comment by rvillablanca** [comment]
ok will see now

**Comment by rvillablanca** [comment]
I don't see any changes in master so I don't have any conflicts :man_shrugging:

**Comment by rvillablanca** [comment]
Sorry just saw the failed CI test, looking into it!

**Comment by rvillablanca** [comment]
Conflicts resolved

**Comment by jasonblais** [comment]
@ogi-m friendly reminder to help with QA review :

**Comment by hanzei** [comment]
@rvillablanca Heads up that there is a merge conflict to resolve

**Comment by rvillablanca** [comment]
Yes, I have a several PRs with conflicts. I am waiting the QA review first so that I solve the conflicts only once and not every time they appears.

We are getting several conflicts because the PRs are stuck in this phase, so it doesn't make sense solving them every now and then.

**Comment by rvillablanca** [comment]
@hanzei some others PRs were approved and this could possibly be approved also, conflicts were solved :)

**Pull request merged**